### PR TITLE
Update to gfx 0.9.1

### LINKED ISSUE #37: Support gfx 0.9
Mostly just raising this issue to let you know I've [made a start](https://github.com/guycook/gfx_text/commit/6756b7199e32d98508ed272e5dc10cbb5dd0e711) to avoid any duplicated effort

### PR DESCRIPTION
**Pull request by guycook**
Fixes #37

This isn't quite ready to merge yet, I think it's ok except for one issue - draw isn't generic over the RenderTargetView format, i.e if in example/style.rs you try to init with an Srgb8 output format it won't build.

I'm not sure what the right approach is here, paging @kvark, @sectopod

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -558,44 +501,34 @@ fn create_texture_r8_static<R: Resources, F: Factory<R>>(
...
-        &texture,
-        &(*texture.get_info()).into(),
-        data,
-        None,
-    ));
-    Ok(texture)
+) -> Result<gfx::handle::ShaderResourceView<R, f32>, CombinedError> {
+    let kind = tex::Kind::D2(width, height, tex::AaMode::Single);
+    let (_, texture_view) = try!(
+        factory.create_texture_const::<(gfx::format::R8, gfx::format::Unorm)>(kind, gfx::cast_slice(&data), false)
```
Comment (kvark) [inline src/lib.rs:507]: I don't think you need a `cast_slice` here, since the format is `(R8, Unorm)` anyway

**Comment by kvark** [comment]
@guycook Nice work!

I see the problem with the output format. There can be several solutions:
1. Have the whole `Renderer` generic over the output format. Is it supposed to be drawing to surfaces of different formats?
2. Have 2 separate functions for `Rgba8` and `Srgba8`, which use different PSOs.
3. Have a map of `output_format -> PSO` in either a typed form (`AnyMap`? Haven't used it so I don't know), or raw form:

``` rust
pub fn draw<T: gfx::format::RenderFormat>(..., target: gfx::RenderTargetView<R, T>) {
    use gfx::core::factory::Typed;
    self.draw_raw(..., target.raw(), T::get_format());
}
fn draw_raw(..., target: gfx::RawRenderTargetView<R>, format: gfx::format::Format) {
    // have a map of `Format -> PSO`, pick the PSO from there and draw
}
```

**Comment by kvark** [comment]
Ping @guycook 
Is this PR dead? gfx-0.10 is out

**Comment by guycook** [comment]
Not dead (in my mind), I just haven't really had the time to attack the problem above due to baby/uni/work etc. Sorry about that.

That said, probably there is no need for this now if gfx 0.10 is out - I would say that if gfx were > 1.0 it would make sense to support minor revisions but while it's 0.\* may as well just track head. Depending on how different 0.9 and 0.10 is maybe this would be a better base to work from than the existing 0.8 code

**Comment by kvark** [comment]
@guycook no worries! You can't just jump right to 0.10 anyway. The difference between 0.9 and 0.10 is small, but you've still got to do the 0.8 to 0.9 step, so your PR needs to be completed in some way.

**Comment by kvark** [comment]
Closing in favor of #41

**Pull request closed without merging**